Import/export saved favorites

huboneo opened the pull request:
This PR heavily refactors the user favorites implementation, and adds support for importing/exporting saved scripts as a ZIP archive

## Key changes
- Replaces local user favorites implementation with a new ui kit package `@relate-by-ui/saved-scripts` (https://github.com/neo4j-apps/relate-by-ui/pull/2)
- Improves appearance and UX of favorites as per design requests
- Updated webpack conf to support symlinked npm packages (for development)
- adds ability to import and export user favorites as .zip archive

## Running this PR locally
- checkout https://github.com/neo4j-apps/relate-by-ui/pull/2
- `yarn install && yarn link` in `packages/saved-scripts`
- checkout this PR
- remove `@relate-by-ui/saved-scripts` from package.json and `yarn install`
- readd `@relate-by-ui/saved-scripts` to package.json 
-`yarn link @relate-by-ui/saved-scripts`
- `yarn start`

- [ ] Run existing tests once new version of `@relate-by-ui/saved-scripts` is published

huboneo commented on src/browser/components/FileDrop/FileDrop.jsx:
```diff
@@ -171,6 +194,10 @@ const mapDispatchToProps = dispatch => {
     },
+    saveManyFavorites: favorites => {
+      const folders = getFoldersFromFavorites(favorites)
+      console.log('saveManyFavorites', favorites, folders)
```
oops, this should be calling `LOAD_FAVORITES` action creator

huboneo replied:
fixed!

huboneo commented on src/browser/modules/Sidebar/Sidebar.jsx:
```diff
@@ -44,83 +45,85 @@ import {
-        content: AboutDrawer
-      }
-    ]
+function Sidebar (props) {
```
Since class only had a render() method anyhow...

huboneo commented on src/shared/modules/favorites/favoritesDuck.js:
```diff
@@ -60,7 +65,7 @@ export default function reducer (state = initialState, action) {
     case LOAD_FAVORITES:
     case UPDATE_FAVORITES:
-      return mergeFavorites(initialState, action.favorites)
+      return mergeFavorites(action.favorites, state)
```
I had to change this to ensure favorites got updated. Still not sure how it worked before, or if this is right

oskarhane commented on build_scripts/webpack-rules.js:
```diff
@@ -98,7 +98,7 @@ module.exports = [
   },
   {
-    test: /\.svg$/,
+    test: /\.(svg|png)/,
```
This break the neo4j logo on `:play start`. And possibly more places.

oskarhane replied:
This is still an issue.
<img width="489" alt="oskarhane-mbpt 2019-09-12 at 08 49 45" src="https://user-images.githubusercontent.com/570998/64760558-50af5c80-d53a-11e9-9fc1-5832ea87af1e.png">

oskarhane commented on package.json:
```diff
@@ -123,27 +123,31 @@
     "firebase": "^5.8.3",
     "isomorphic-fetch": "^2.2.1",
     "jsonic": "^0.3.0",
+    "jszip": "^3.2.2",
```
We need to verify that this works in IE 11.

oskarhane commented on src/browser/components/FileDrop/file-drop.utils.test.js:
```diff
@@ -0,0 +1,130 @@
+  getMissingFoldersFromNames
+} from './file-drop.utils'
+
+describe('file-drop.utils', () => {
```
👍

huboneo commented:
@oskarhane I've updated the PR to address 1. I also found two unrelated bugs from the browser-lambda-parser PR which I fixed. Please check my changes to build_scripts so that I didn't miss anything

oskarhane commented on build_scripts/webpack-rules.js:
```diff
@@ -23,7 +23,11 @@ const path = require('path')
   {
     test: /\.(js|jsx)$/,
-    exclude: /(node_modules)|(cypher-codemirror)|(test_utils)|(dist)/,
+    include: [
```
I'm a bit worried that this might break something but from my tests it all seems fine.
We should verify each step in the build pipeline after this is merged.

huboneo commented:
@oskarhane Sorry completely forgot about the webpack bug. Fixed:
<img width="1256" alt="Screenshot 2019-09-12 at 09 25 51" src="https://user-images.githubusercontent.com/52443771/64763004-6ecb8b80-d53f-11e9-84f3-7f1ac5a9b646.png">
